dotcom-rendering/src/types/discussion.ts

```diff
@@ -141,11 +145,60 @@ export interface CommentType {
 	};
 }
 
-export type CommentResponse = {
-	status: 'ok' | 'error';
-	statusCode: number;
-	message: string;
-	errorCode?: string;
+export type CommentResponseErrorCodes = (typeof errorCodes)[number];
+const errorCodes = [
+	'USERNAME_MISSING',
+	'EMPTY_COMMENT_BODY',
+	'COMMENT_TOO_LONG',
+	'USER_BANNED',
+	'IP_THROTTLED',
+	'DISCUSSION_CLOSED',
+	'PARENT_COMMENT_MODERATED',
+	'COMMENT_RATE_LIMIT_EXCEEDED',
+	'INVALID_PROTOCOL',
+	'AUTH_COOKIE_INVALID',
+	'READ-ONLY-MODE',
+	'API_CORS_BLOCKED',
+	'API_ERROR',
+	'EMAIL_VERIFIED',
+	'EMAIL_VERIFIED_FAIL',
+	'EMAIL_NOT_VALIDATED',
+] as const;
+
+const commentErrorSchema = object({
+	status: literal('error'),
+	errorCode: picklist(errorCodes),
+	message: string(),
+});
+
+const commentResponseSchema = variant('status', [
+	commentErrorSchema,
+	object({
+		status: literal('ok'),
+		// response.errorCode is the id of the comment that was created on the server
+		// it is returned as a string, so we need to cast to an number to be compatible
+		message: transform(string(), (input) => parseInt(input, 10)),
```

Comment:
What if `NaN`?

Reply:
Fixed in 9641ca55acc31b9f9346fc2a0408200af22d7f43